docs/guides/sqlite3_db_with_concord.md

```diff
@@ -0,0 +1,259 @@
+# Other topics
+
+## Using a DB with concord
+
+Sometimes, when you are developing a command, you may wish to save some data, that will be not deleted across the restarts from your bot, this is where a database comes in. It can save data, and, don't worry, it will not be deleted if you restart your bot.
+
+In this guide, we will be teaching you how to use sqlite3, a local database, that it's incredibly fast, easy to use and debug.
+
+### Opening the DB file
+
+The first step in this guide will be how to open the DB file, as it's required for the other sqlite3 functions like read and write data.
+
+```c
+#include <sqlite3.h>
+
+...
+
+sqlite3 *db;
+int rc = sqlite3_open("db.sqlite", &db);
+
+if (rc != SQLITE_OK) {
+    log_fatal("[SQLITE] Error when opening the db file. [%s]", sqlite3_errmsg(db));
+      
+    if (sqlite3_close(db) != SQLITE_OK) {
+        log_fatal("[SQLITE] Failed to close sqlite db. [%s]", sqlite3_errmsg(db));
+        abort();
+    }
+    return;
+}
+```
+
+In the first line, we are simply including the sqlite3 into the file.
+
+Then, we are **opening the DB file** with `sqlite3_open` and name it db.sqlite (NOTE: If the file doesn't exist, sqlite3 will create it for you). After opening, it will **check if the status of the `sqlite3_open` is not equal to `SQLITE_OK`** (that represents that it was successfully executed), it's not equal, it will log a fatal error about it, getting the error message with `sqlite3_errmsg` and then closing the DB with `sqlite3_close`. (NOTE: **Yes, even failing to open, you MUST use `sqlite3_close` as said in sqlite3 docs**)
+
+> Whether or not an error occurs when it is opened, resources associated with the database connection handle should be released by passing it to sqlite3_close() when it is no longer required.
+
+Rarely this happens, but if the `sqlite3_close` **fails to be executed**, we will be just killing the bot's process. Since this is not a highly detailed guide, we won't be teaching how to deal with it with many details, but you will only need to finish the finalize the ongoing statement and then use `sqlite3_close` again.
```

Comment:
"Rarely this happens, but if the `sqlite3_close` **fails to be executed**, we will be just killing the bot's process. Since this is not a highly detailed guide, we won't be teaching how to deal with it with great detail; you will only need to finish the finalize the ongoing statement and then use `sqlite3_close` again."

Reply:
Mhm, thanks